[Bug]: Check sphinx links workflow failing because of new sphinx warning

### What happened?

The sphinx link checking GitHub action is currently failing.

I think this is because of a new sphinx warning. Further discussion of this issue and the steps to address are here: https://github.com/sphinx-doc/sphinx/issues/12300 

### Steps to Reproduce

```python
Run GitHub workflow
```


### Traceback

```python
Warning, treated as error:
cannot cache unpickable configuration value: 'sphinx_gallery_conf' (because it contains a function, class, or module object)
Error: Process completed with exit code 2.
```


### Operating System

macOS

### Python Executable

Conda

### Python Version

3.11

### Package Versions

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/.github/CODE_OF_CONDUCT.rst)
- [X] Have you checked the [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [X] Have you ensured this bug was not already [reported](https://github.com/NeurodataWithoutBorders/pynwb/issues)?

--- a/docs/source/conf.py
+++ b/docs/source/conf.py
@@ -168,6 +168,8 @@ nitpicky = True
 nitpick_ignore = [('py:class', 'Intracomm'),
                   ('py:class', 'BaseStorageSpec')]
 
+suppress_warnings = ["config.cache"]
+
 # Add any paths that contain templates here, relative to this directory.
 templates_path = ['_templates']
 

